File: 8th-April-Make_the_array_beautiful/c++/solution.cpp

class Solution {
  public:
    vector<int> makeBeautiful(vector<int> arr) {
        // code here
        vector<int> list;
        list.push_back(arr[0]);
    
    for(int i = 1; i < arr.size(); i++) {
        if(list.empty()){
            list.push_back(arr[i]);
        } 
        else{
            if(list.back() >= 0 && arr[i] >= 0) {
                list.push_back(arr[i]);
            } 
            else if(list.back() < 0 && arr[i] < 0) {
                list.push_back(arr[i]);
            } 
            else {
                list.pop_back();
            }
        }
    }
    
    return list;

    }
};

File: 8th-April-Make_the_array_beautiful/java/solution.java
class Solution {
    public static ArrayList<Integer> makeBeautiful(int[] arr) {
        // code here
        ArrayList<Integer> list = new ArrayList<>();
        
        list.add(arr[0]);
        
        for(int i =1 ; i< arr.length ; i++){
            if(list.isEmpty()){
                list.add(arr[i]);
            }
            else
            {
                if(list.get(list.size()-1) >=0 && arr[i] >=0 ){
                    list.add(arr[i]);
                }
                else if(list.get(list.size()-1) < 0 && arr[i] < 0 ){
                    list.add(arr[i]);
                }
                else{
                    list.remove(list.size()-1);
                }
            }
        }
        return list;
    }
}
